examples/specialty_plots/ishikawa.py

```diff
@@ -0,0 +1,229 @@
+        The default is [np.nan].
+    spine_color : str, optional
+        Spine color. The default is "black".
+    reclevel : int, optional
+        Current recursion level. The default is 1.
+    max_recursion_level : int, optional
+        Maximum recursion level set. The default is 4.
+    debug : bool, optional
+        Print debug variables. The default is False.
+
+    Raises
+    ------
+    AttributeError
+        Maximum recursion level reached or passed a bad parentspine object format.
+
+    Returns
+    -------
+    None.
+
+    """
+    # stop recursion if maximum level reached
+    if reclevel > max_recursion_level:
+        raise AttributeError('Max Recursion Level Reached')
+
+    if isinstance(data, dict):
+        # switch to correct angle depending on recursion level
+        if reclevel % 2 != 0:
+            alpha = alpha_ps
+        else:
+            alpha = alpha_ss
+
+        if isinstance(parentspine, matplotlib.lines.Line2D):
+            # calculate parent data
+            ([xpb, xpe], [ypb, ype]) = parentspine.get_data()
+        elif isinstance(parentspine, matplotlib.text.Annotation):
+            xpb, ypb = parentspine.xy
+            xpe, ype = parentspine._x, parentspine._y  # parentspine._arrow_relpos
+        else:
+            raise AttributeError('Wrong Spine Graphical Element')
+
+        plen = np.sqrt((xpe - xpb) ** 2 + (ype - ypb) ** 2)
+        palpha = np.round(np.degrees(np.arctan2(ype - ypb, xpe - xpb)))
+
+        # calculate spine spacing
+        # calculate couple pairs, at least 1 pair to start at middle branch
+        pairs = len(list(data.keys())) // 2 + len(list(data.keys())) % 2 + 1
+        spacing = plen / pairs
+
+        spine_count = 0
+        s = spacing
+        # draw spine
+        for k in data.keys():
+            # calculate arrow position in the graph
+            # fix primary spines spacing
+            if reclevel == 1:
+                if len(primary_spines_rel_xpos) == len(list(data.keys())):
+                    x_end = primary_spines_rel_xpos[spine_count]
+                else:
+                    x_end = xpe - (s / 1.5) * np.cos(np.radians(palpha))
+                y_end = ype - s * np.sin(np.radians(palpha))
+
+                x_start = x_end - s * np.cos(np.radians(alpha))
+                y_start = y_end - s * np.sin(np.radians(alpha))
+            else:
+                x_end = xpe - s * np.cos(np.radians(palpha))
+                y_end = ype - s * np.sin(np.radians(palpha))
+
+                x_start = x_end - s * np.cos(np.radians(alpha))
+                y_start = y_end - s * np.sin(np.radians(alpha))
+            if debug:
+                _log.debug(
+                    f'k: {k}, x_start: {x_start}, y_start: {y_start}, '
+                    f'x_end: {x_end} y_end: {y_end}, alpha: {alpha} \n '
+                    f'reclevel: {reclevel}, '
+                    f's:{s}, plen:{plen}, palpha:{palpha}')
```

Comment:
f-strings should not be used in logging, cf #25081. Also, the point of using logging is you don't need `if debug:` everywhere.